appengine/standard/firebase/firetactoe/firetactoe.py

```diff
@@ -0,0 +1,288 @@
+      
+    for win in wins:
+      if win.match(self.game.board):
+        self.game.winner = potential_winner
+        self.game.winning_board = win.pattern
+        return
+
+  def make_move(self, position, user):
+    if position >= 0 and user == self.game.userX or user == self.game.userO:
+      if self.game.moveX == (user == self.game.userX):
+        boardList = list(self.game.board)
+        if (boardList[position] == ' '):
+          boardList[position] = 'X' if self.game.moveX else 'O'
+          self.game.board = "".join(boardList)
+          self.game.moveX = not self.game.moveX
+          self.check_win()
+          self.game.put()
+          self.send_update()
+          return
+
+
+class GameFromRequest():
+  game = None;
+
+  def __init__(self, request):
+    user = users.get_current_user()
+    game_key = request.get('g')
+    if user and game_key:
+      self.game = Game.get_by_key_name(game_key)
+
+  def get_game(self):
+    return self.game
+
+
+class MovePage(webapp.RequestHandler):
+
+  def post(self):
+    game = GameFromRequest(self.request).get_game()
+    user = users.get_current_user()
+    if game and user:
+      id = int(self.request.get('i'))
+      GameUpdater(game).make_move(id, user)
+
+class DeletePage(webapp.RequestHandler):
+
+  def post(self):
+    game = GameFromRequest(self.request).get_game()
+    user = users.get_current_user()
+    if game and user:
+      FireChannel(fire_url).send_message(u_id=user.user_id() + game.key().id_or_name(), 
+    									message=None,
+    									is_delete=True)
+
+class OpenedPage(webapp.RequestHandler):
+  def post(self):
+    game = GameFromRequest(self.request).get_game()
+    GameUpdater(game).send_update()
+
+# This class has a set of helper functions on top of url fetch to make it easier
+# to send commands to Firebase. It consists of a constructor that saves the
+# Firebase database URL. In addition, it implements send_message, which updates
+# a path on Firebase based on the unique identifier u_id. In this example
+# u_id == channel_id == user_id + game_key
+class FireChannel():
+	base_url = None
+
+	def __init__(self, base_url):
+		self.base_url = base_url
+
+	def send_message(self, u_id, message, is_delete):
+		try:
+			headers = {'Content-Type': 'application/x-www-form-urlencoded'}
+			method_type = None
+			if (is_delete): 
+				method_type = urlfetch.DELETE 
+			else: 
+				method_type = urlfetch.PATCH
+			# always using patch in this example for writes
+			# because it updates or creates but never overwrites
+			result = urlfetch.fetch(
+        		url=self.base_url + u_id + '.json?auth=' + gae_auth_token, # create a new stream using unique id and server auth key
+        		payload=message,
+        		method=method_type,
+        		headers=headers,
+        		validate_certificate=True)
+			json_data = simplejson.loads(result.content)
+			logging.info(json_data.get("message"))
+			return result.content
+		except urlfetch.Error:
```

Comment:
In samples, it's better to let the exception raise all the way up so that the user can see the stacktrace. This helps us figure out what's wrong if they report issues.